### PR TITLE
Add keyboard meetlab/kalice

### PR DESCRIPTION
**Pull request by LXF-YZP**
## Description
Add keyboard meetlab/kalice

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/meetlab/kalice/alice.json
Comment (zvecr) [inline keyboards/meetlab/kalice/alice.json:1]: This file must be removed.

File: keyboards/meetlab/kalice/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		    KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS),
+
+[3] = LAYOUT(
+   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, 
+		    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		    KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS),
+
```
Comment (zvecr) [inline keyboards/meetlab/kalice/keymaps/via/keymap.c:48]:
```suggestion
		    KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS)
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/meetlab/kalice/alice.json
Comment (LXF-YZP) [reply keyboards/meetlab/kalice/alice.json:1]: Sorry, forgot to remove it

File: keyboards/meetlab/kalice/info.json [hunk 2]
```diff
@@ -0,0 +1,128 @@
+{
+    "manufacturer": "lucky_studio",
+    "keyboard_name": "Kalice",
+    "maintainer": "yuezp",
+    "bootloader": "stm32duino",
```
Comment (drashna) [inline keyboards/meetlab/kalice/info.json:5]: You can replace the bootloader and process with just: 
```suggestion
    "development_board": "bluepill",
```

File: keyboards/meetlab/kalice/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[0] = LAYOUT(
+   KC_ESC,  KC_GRV,  KC_1,    KC_2,    KC_3,   KC_4,    KC_5,    KC_6,    KC_7,    KC_8,     KC_9,     KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_INS, 
+   KC_PGUP, KC_TAB,  KC_Q,    KC_W,    KC_E,   KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,     KC_O,     KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_DEL, 
+   KC_PGDN, KC_CAPS, KC_A,    KC_S,    KC_D,   KC_F,    KC_G,    KC_H,    KC_J,    KC_K,     KC_L,     KC_SCLN, KC_QUOT,          KC_ENT,   MO(1), 
+		    KC_LSFT, KC_Z,    KC_X,    KC_C,   KC_V,    KC_B,    KC_B,    KC_N,    KC_M,     KC_COMM,  KC_DOT,  KC_SLSH, KC_RSFT, KC_UP, 
+		    KC_LCTL, KC_LGUI, KC_LALT,         KC_SPC,                    KC_SPC,  KC_RCTL,  KC_RALT,                    KC_LEFT, KC_DOWN,  KC_RGHT),
```
Comment (drashna) [inline keyboards/meetlab/kalice/keymaps/default/keymap.c:26]:
```suggestion
    KC_LSFT, KC_Z,    KC_X,    KC_C,   KC_V,    KC_B,    KC_B,    KC_N,    KC_M,     KC_COMM,  KC_DOT,  KC_SLSH, KC_RSFT, KC_UP, 
    KC_LCTL, KC_LGUI, KC_LALT,         KC_SPC,                    KC_SPC,  KC_RCTL,  KC_RALT,                    KC_LEFT, KC_DOWN,  KC_RGHT),
```

File: keyboards/meetlab/kalice/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[0] = LAYOUT(
+   KC_ESC,  KC_GRV,  KC_1,    KC_2,    KC_3,   KC_4,  KC_5,   KC_6,    KC_7,    KC_8,    KC_9,    KC_0,     KC_MINS, KC_EQL,  KC_BSPC, KC_INS, 
+   KC_PGUP, KC_TAB,  KC_Q,    KC_W,    KC_E,   KC_R,  KC_T,   KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,     KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL, 
+   KC_PGDN, KC_CAPS, KC_A,    KC_S,    KC_D,   KC_F,  KC_G,   KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,  KC_QUOT,          KC_ENT,  MO(1), 
+		    KC_LSFT, KC_Z,    KC_X,    KC_C,   KC_V,  KC_B,   KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,   KC_SLSH, KC_RSFT, KC_UP, 
+		    KC_LCTL, KC_LGUI, KC_LALT,         KC_SPC,                 KC_SPC,  KC_RCTL, KC_RALT,                    KC_LEFT, KC_DOWN, KC_RGHT),
```
Comment (drashna) [inline keyboards/meetlab/kalice/keymaps/via/keymap.c:26]:
```suggestion
    KC_LSFT, KC_Z,    KC_X,    KC_C,   KC_V,    KC_B,    KC_B,    KC_N,    KC_M,     KC_COMM,  KC_DOT,  KC_SLSH, KC_RSFT, KC_UP, 
    KC_LCTL, KC_LGUI, KC_LALT,         KC_SPC,                    KC_SPC,  KC_RCTL,  KC_RALT,                    KC_LEFT, KC_DOWN,  KC_RGHT),
```

File: keyboards/meetlab/kalice/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+[1] = LAYOUT(
+   KC_TRNS, KC_TRNS, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS, KC_TRNS, 
+   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, 
+		    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		    KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS)
```
Comment (drashna) [inline keyboards/meetlab/kalice/keymaps/default/keymap.c:33]:
```suggestion
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
    KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS)
```

File: keyboards/meetlab/kalice/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+[1] = LAYOUT(
+   KC_TRNS, KC_TRNS, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS, KC_TRNS, 
+   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS, 
+		    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		    KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS)
```
Comment (drashna) [inline keyboards/meetlab/kalice/keymaps/via/keymap.c:33]:
```suggestion
    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
    KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS)
```

File: keyboards/meetlab/kalice/info.json (hunk 2 above)
Comment (LXF-YZP) [reply keyboards/meetlab/kalice/info.json:5]: thanks

File: keyboards/meetlab/kalice/keymaps/default/keymap.c (hunk 3 above)
Comment (LXF-YZP) [reply keyboards/meetlab/kalice/keymaps/default/keymap.c]: I'm sorry for the trouble caused by the format problem

**Review by drashna — APPROVED** [review]

**Comment by LXF-YZP** [comment]
Excuse me, is there anything I need to change? @drashna @zvecr

File: keyboards/meetlab/kalice/halconf.h [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include_next <halconf.h>
+
+#undef HAL_USE_PWM
+#define HAL_USE_PWM    TRUE
```
Comment (fauxpark) [inline keyboards/meetlab/kalice/halconf.h:21]:
```suggestion

#define HAL_USE_PWM TRUE

#include_next <halconf.h>
```

File: keyboards/meetlab/kalice/info.json
Comment (fauxpark) [inline keyboards/meetlab/kalice/info.json:1]: Please run `qmk format-json` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by LXF-YZP** [comment]
hello, Can you check it for me? Thank you。 @zvecr

**Review by fauxpark — APPROVED** [review]

File: keyboards/meetlab/kalice/alice.json
Comment (LXF-YZP) [reply keyboards/meetlab/kalice/alice.json:1]: > This file must be removed.

hello, Can you check it for me? Thank you。 @zvecr

**Comment by LXF-YZP** [comment]
What can I do for this PR else? Would you please merge it now?  @waffle87 @drashna @fauxpark

**Pull request merged**